onnx_caffe2/helper.py

```diff
@@ -60,17 +68,65 @@ def c2_native_run_net(init_net, predict_net, inputs):
 
     if isinstance(inputs, dict):
         for key, value in inputs.items():
-            ws.FeedBlob(key, value)
+            ws.FeedBlob(key, value, predict_net.device_option)
     else:
         uninitialized = [input_name
                          for input_name in predict_net.external_input
                          if not ws.HasBlob(input_name)]
         assert len(uninitialized) == len(inputs)
         for key, value in zip(uninitialized, inputs):
-            ws.FeedBlob(key, value)
+            ws.FeedBlob(key, value, predict_net.device_option)
 
     ws.RunNetOnce(predict_net)
 
     output_names = predict_net.external_output
     output_values = [ws.FetchBlob(name) for name in output_names]
     return ws, namedtupledict('Outputs', output_names)(*output_values)
+
+
+def name_inputs(onnx_model, inputs):
```

Comment:
you should have the names of blobs in predict_net.external_input. So maybe just pass the list directly to c2_native_run_net?

Reply:
init_net contains the fake input. So the assertion len(uninitialized) == len(inputs) fails. So it's better to pass a dict.

Reply:
What I meant is to accept the list directly in c2_native_run_net() and put this code there